src/redis_ai_objects/tensor.c

```diff
@@ -651,6 +651,7 @@ int RAI_TensorSetData(RAI_Tensor *t, const char *data, size_t len) {
         if (_RAI_TensorParseStringsBlob(data, len, RAI_TensorLength(t),
                                         RAI_TensorStringElementsOffsets(t),
                                         &err) != REDISMODULE_OK) {
+            RAI_ClearError(&err);
```

Comment:
To avoid leaks. If we get an error in parsing the blob, the error message is allocated. We use this message when calling `_RAI_TensorParseStringsBlob` in `AI.TENSORSET` command path.

Reply:
but why are you clearing the error in case of non-successful execution?

Reply:
note to myself - send NULL as `err` and handle this case in `_RAI_TensorParseStringsBlob`